### PR TITLE
rebased, and added functionality to Add Class & ClassWork

### PR DESCRIPTION
**Pull request by Marcellia**
#77 and #47 

Connected to backend to fetch data for Class & classWork, and ability to add new data for Class

Screen Shot 2022-06-18 at 7 58 57 AM
Screen Shot 2022-06-18 at 7 58 38 AM

### CODE REVIEW

File: src/pages/Admin/AdminUpdateClass/AdminUpdateClass.tsx [hunk 1]
```diff
@@ -12,11 +12,42 @@ import {
...
+  const { id } = useParams();
+
+  const fetchAdminClasses = useCallback(async () => {
+    try {
+      const res = await getAdminClass(id);
+      setClasses(res.data);
+    } catch (error) {
+      const errors = error as ErrorMessageType[];
+      if (errors?.[0]?.msg === ERROR_MESSAGES.unauthorized) {
+        // navigate('/sign-in', { replace: true });
```
Comment (jpatrici) [inline src/pages/Admin/AdminUpdateClass/AdminUpdateClass.tsx:41]: you can delete this and replace it with any other error handling :) I believe this part should be covered by the session cookie later on

File: package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "react-modal": "^3.15.1",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
+    "react-uid": "^2.3.2",
```
[the remark this answers is not in the document]
Comment (Marcellia) [reply package.json:28]: updated

File: src/utilities/dateFormat.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export const getFormattedDate = (dateString: string) => {
+  const date = new Date(dateString);
+  const day = date.getUTCDate();
+  const month = date.getUTCMonth();
+  const year = date.getUTCFullYear();
+
+  return new Intl.DateTimeFormat(lang, options).format(
+    new Date(year, month, day)
+  );
+};
```
[the remark this answers is not in the document]
Comment (Marcellia) [reply src/utilities/dateFormat.tsx:17]: yes 💯

File: src/pages/Admin/AdminUpdateClass/AddNewClassModal.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  AdminCreateClassProps,
+} from '../../../api/getAdminClasses';
+
+import {
+  FormErrorMessages,
+  StyledErrorMessage,
+} from '../../../components/ErrorMessage';
+
+import { Modal } from '../../../components/Modal/Modal';
+import { Button, PrimaryButton } from '../../../components/Button';
```
Comment (jpatrici) [inline src/pages/Admin/AdminUpdateClass/AddNewClassModal.tsx:18]: hmm i'll create an index.tsx file later to clean these imports a bit

**Review by jpatrici — APPROVED** [review]
🚀

**Pull request merged**